Gems/Terrain/Assets/Shaders/Terrain/ClipmapComputeHelpers.azsli

```diff
@@ -229,10 +241,15 @@ BilinearUvs CalculateClipmapUv(
     return uvs;
 }
 
-float4 ColorPointSampling(Texture2DArray<float4> colorClipmap, uint2 uv, uint clipmapLevel)
+float4 ColorPointSampling(Texture2DArray<float4> colorClipmap, uint2 uv, uint clipmapLevel, float debugEnabled)
 {
     uint3 texelIndex = uint3(uv, clipmapLevel);
-    return colorClipmap[texelIndex];
+    float4 color = colorClipmap[texelIndex];
+    // Debug option
+    float3 overlayColor = ClipmapLevelDebugColor[min(clipmapLevel, 4)];
+    color.xyz += (color.xyz * -0.25 + overlayColor * 0.25) * debugEnabled;
```

Comment:
In general I like to see debugging stuff behind shader options but I _think_ it's ok here since you're passing in 0.0 from the non-debug pass. I'd expect the shader compiler to optimize it away in that case.

Reply:
Yeah. This is a fairly low cost debug rendering, it should save more on shader permutations.